### PR TITLE
Ability to Login

### PR DESCRIPTION
**Pull request by AinoraZ**
Changes:
  - Added ability to login.
  - Some routes are now private routes (routed back to /login if no user is stored).
  - Added UserContextManager to handle login/logout/ping for users.

Related style changes:
  - Removed black border from navigation bar, as it was showing up even when navigation is empty and was barely visible when it wasn't empty.
  - Removed min-height from content class because it was causing weird issues. Comment on this, if there was a specific reason for it existing. Didn't notice anything breaking.

Known issues:
  - Couldn't remove the navbar completely because of the way the project is structured currently. Also had a lot of issues with bootstrap and overall content because of current css rules which don't play well with bootstrap. For next iteration, I suggest switching to a bootstrap friendly navbar/sidebar/content by wrapping everything in a bootstrap container and using grids to tell how much space should be taken up by the navbar, sidebar and content. This would also have the benefit of scaling correctly when switching to smaller screens.

### CODE REVIEW

**Comment by AinoraZ** [comment]
[Task 4: Create Login page](https://trello.com/c/V50Ayr20/56-task-4-create-login-page)

File: billchop-fe/src/App.tsx [hunk 1]
```diff
@@ -1,37 +1,59 @@
...
 import CreateGroupPage from "./pages/CreateGroupPage";
 import "./App.css";
 import NavigationBar from "./components/NavigationBar";
+import { PrivateRoute } from "./PrivateRoute";
+import BrowserHistory from "./backend/helpers/History";
+import User from "./backend/models/User";
+import UserContext from "./backend/helpers/UserContext";
+import LoginPage from "./pages/LoginPage";
+
+export interface AppState {
```
Comment (tmaciulis22) [inline billchop-fe/src/App.tsx]: its a minor thing and dunno whether we should stick to this convention, but I guess interfaces should start with `I`

File: billchop-fe/src/components/NavigationBar.tsx [hunk 2]
```diff
@@ -5,18 +5,27 @@ import Avatar from "react-avatar";
...
-      <Navbar bg="light">
+      <Navbar className="mainContainer__pageHeader" bg="light">
         <Navbar.Toggle aria-controls="basic-navbar-nav" />
         <Navbar.Collapse id="basic-navbar-nav">
           <Nav className="mr-auto">
             <Link to="/profile">
-              {/* Pass username as props */}
               <div className="mr-2">
-                <Avatar name="Username" round size="40" />
+                <Avatar name={`${currentUser.Name}`} round size="40" />
```
Comment (tmaciulis22) [inline billchop-fe/src/components/NavigationBar.tsx]: wouldn't `name={currentUser.Name}` do the same here?

File: billchop-fe/src/pages/LoginPage.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { Form, Button, Row, Col, Container } from "react-bootstrap";
+import UserContext from "../backend/helpers/UserContext";
+import BrowserHistory from "../backend/helpers/History";
+import "./LoginPage.scss";
+
+interface LoginPageState {
```
Comment (tmaciulis22) [inline billchop-fe/src/pages/LoginPage.tsx]: naming, but I guess its fine this way as well xd

**Review by tmaciulis22 — DISMISSED** [review]

**Review by tmaciulis22 — APPROVED** [review]

**Pull request merged**